toolkit/tools/internal/network/network.go

```diff
@@ -30,14 +31,23 @@ func JoinURL(baseURL string, extraPaths ...string) string {
 	return fmt.Sprintf("%s%s%s", baseURL, urlPathSeparator, appendToBase)
 }
 
-// DownloadFile downloads `url` into `dst`. `caCerts` may be nil.
+// DownloadFile downloads `url` into `dst`. `caCerts` may be nil. If there is an error `dst` will be removed.
 func DownloadFile(url, dst string, caCerts *x509.CertPool, tlsCerts []tls.Certificate) (err error) {
 	logger.Log.Debugf("Downloading (%s) -> (%s)", url, dst)
 
 	dstFile, err := os.Create(dst)
 	if err != nil {
 		return
 	}
+	defer func() {
+		// If there was an error, ensure that the file is removed
+		if err != nil {
+			cleanupErr := removeFileIfExists(dst)
+			if cleanupErr != nil {
+				logger.Log.Errorf("Failed to remove failed network download file '%s': %s", dst, err)
+			}
+		}
+	}()
 	defer dstFile.Close()
```

Comment:
Now the file will get closed before it gets removed, so there's a window where another thread/process may create a new file, which gets then removed by this code. I'd reverse the order of operations. Also, I think there's no need for two separate `defer` if they are called immediately one after another - we can lump these together.